core/sql/ustat/hs_globals.cpp

```diff
@@ -546,10 +557,11 @@ NABoolean HSGlobalsClass::setHBaseCacheSize(double sampleRatio)
   Int64 workableCacheSize = (Int64)(sampleRatio * calibrationFactor);
   if (workableCacheSize < 1)
     workableCacheSize = 1;  // can't go below 1 unfortunately
+  else if (workableCacheSize > 50)
+    workableCacheSize = 50; 
 
-  Int32 max = getDefaultAsLong(HBASE_NUM_CACHE_ROWS_MAX);
-  if ((workableCacheSize < 10000) && // don't bother if 10000 works
-      (max == 10000))  // don't do it if user has already set this CQD
+  Int32 maxDefault = getDefaultAsLong(HBASE_NUM_CACHE_ROWS_MAX);
+  if (maxDefault == 10000) // don't do it if user has already set this CQD
```

Comment:
Thanks for the suggestion for NADefaults::getProvenance(). I'll make this change.

In the case that the user sets the CQD him/herself, I wanted to give them complete freedom to experiment as they please. Higher values may make UPDATE STATS run faster (though experiements suggest not by much). The trade-off in using higher values is an increased risk of failure.